LabApp_1/LabApp_1/ViewModelContent/FiguresConverter.cs

```diff
@@ -0,0 +1,205 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Windows;
+using System.Windows.Data;
+
+namespace LabApp
+{
+    class FiguresConverter : IValueConverter
+    {
+        public FiguresConverter()
+        {
+
+        }
+
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            ObservableCollection<System.Windows.Shapes.Shape> res = new ObservableCollection<System.Windows.Shapes.Shape>();
+
+            for (int i = 0; i < (value as FiguresObservableCollection).Count; ++i)
+                res.Add(CreateCorrespondingShape((value as FiguresObservableCollection)[i]));
+
+            return res;
+        }
+
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            throw new NotImplementedException();
+        }
+
+        public static System.Windows.Shapes.Shape CreateCorrespondingShape(AbstractFigure figure)
+        {
+            int id = figure.GetID();
+            System.Windows.Shapes.Shape shape = null;
+            switch (id)
```

Comment:
this is violation of the open-closed principle in solid. Every time you need to add a new figure you would change the implementation. That's not correct